### DISCUSSION TITLE
Adding Prefix to a particular endpoint

### DISCUSSION THREAD

**Discussion by VarunS5** [serverpod/serverpod#2103] [Q&A]
How can we add particular prefix path to a endpoint for example :

class ExampleEndpoint extends Endpoint {
   Future<String> hello(Session session){
       return "hello";
   }
}

the rest api path as per code will be example/hello,
There is a requirement to have prefix like "custom/example/hello"
How to modify code to add the **custom** on the prefix?

**Comment by Isakdl** [comment]
Not supported today, but would be a nice feature to add, I'm converting this into an issue!

**Reply by Isakdl** [reply]
@VarunS5 With that said you are able to use the web server to specify any path you like, but with the drawback that you will need to make the client requests yourself (I.E. there is no generated endpoints in the client). https://docs.serverpod.dev/concepts/webserver

**Reply by VarunS5** [reply]
@Isakdl Thanks for the reply, As of now I am solving this issue by the following methods
class Custom extends Endpoint {
  final idsHub = IdsHub();

  Future<Map<String, dynamic>> example(Session session) async {
    final requestedPath = (session as MethodCallSession)
        .httpRequest
        .requestedUri
        .path
        .split("/")
        .last;
    if (requestedPath == "hello") {
      // logic for hello
    }
    if (requestedPath == "something") {
      // logic for something
    } else {
      return {"error": "method not found"};
    }
  }
}

**Reply by Isakdl** [reply]
I see, may I ask what your use case is? Why do you need to implement your own routing logic and why do you need a specific path?

While your implementation may work we do not give any guarantees for this usage in the current API will continue to function in the future.

**Reply by VarunS5** [reply]
Actually the use case we are using the serverpod to create a mock server for our development, we chose this because of the ease of development, and some mock urls have similar path but domain is different, and want to have different prefix to differentiate

**Reply by VarunS5** [reply]
This is completely a workaround, but in future we may need this feature
